File: 7-Grafos/1194-Prefixa_Infixa_e_Posfixa-3/1194.c
#include <stdio.h>
#include <string.h>
int p_i(char infixa[], char proc, int size) {
    for (int i = 0; i < size; i++) {
        if (infixa[i] == proc) {
            return i - 1;
        }
    }
    return -1;
}

int main() {
    int repeat;
    scanf("%d", &repeat);

    for (int r = 0; r < repeat; r++) {
        int qtd;
        char prefixa[510], infixa[510];
        scanf("%d %s %s", &qtd, prefixa, infixa);

        char raiz = prefixa[0];
        char result[510] = "";
        
        int qtd2 = p_i(infixa, prefixa[0], qtd);

        int s;
        if(qtd2==0){
            s=0;
        }
        else{
            s=1;
        }

        int c = 0;
        int passed = 1;
        int passed2 = 1;
        int nad = 0;


        for (int i = 0; i < qtd; i++) {
            if (nad) {
                if (passed) {
                    c++;
                    passed = 0;
                } else {
                    qtd--;
                    passed = 1;
                    c++;
                }
            } else if (infixa[c] == raiz) {
                nad = 1;
                c++;
            } else {
                if (passed2) {
                    strncat(result, &infixa[c], 1); 
                    passed2 = 0;
                    c++;
                } else {
                    if (infixa[qtd2 - 1] == infixa[c]) {
                        strncat(result, &infixa[qtd2], 1);
                        strncat(result, &infixa[qtd2 - 1], 1);
                        break;
                    } else {
                        strncat(result, &infixa[qtd2], 1);
                    }
                    passed2 = 1;
                    c++;
                }
            }
        }

        passed = 1;
        qtd = strlen(infixa);
        nad = 0;
        c = 0;

        for (int i = 0; i < qtd+s; i++) {
            if (nad) {
                if (passed) {
                    strncat(result, &infixa[c], 1);
                    c++;
                    passed = 0;
                } else {
                    strncat(result, &infixa[qtd - 1], 1);
                    qtd--;
                    passed = 1;
                }
            } else if (infixa[c] == raiz) {
                nad = 1;
                c++;
            } else {
                c++;
            }
        }
        strncat(result, &raiz, 1);
        printf("%s\n", result);
    }

    return 0;
}

File: Beecrowd/7-Grafos/1194-Prefixa_Infixa_e_Posfixa-3/1194.py
#Resolucao Porca, e incorreta, ma criacao de casos de teste

repeat=int(input())

def p_i(infixa, proc, size) -> int:
    for i in range(0, size, +1):
        if infixa[i]==proc:
            return i-1

    
for r in range(0, repeat, +1):
    data=input().split(' ')
    prefixa=data[1]
    infixa=data[2]
    
    raiz=prefixa[0]
    result=""

    passed=True

    qtd=int(data[0])
    nad=False

    qtd2=p_i(infixa, prefixa[0], qtd)

    c=0
    passed2=True

    for i in range(0, int(data[0]), +1):
        if nad:
            if passed:
                c+=1
                passed=False
            else:
                qtd-=1
                passed=True
                c+=1
        elif infixa[c] == raiz:
            nad=True
            c+=1
        else:
            if passed2:
                result+=infixa[c]
                passed2=False
                c+=1
            else:
                if infixa[qtd2-1]==infixa[c]:
                    result+=infixa[qtd2]+infixa[qtd2-1]
                    break
                else:
                    result+=infixa[qtd2]
            
                passed2=True
                c+=1

    passed=True

    qtd=int(data[0])
    nad=False

    c=0

    for i in range(0, int(data[0]), +1):
        if nad:
            if passed:
                result+=infixa[c]
                c+=1
                passed=False
            else:
                result+=infixa[qtd-1]
                qtd-=1
                passed=True
        elif infixa[c] == raiz:
            nad=True
            c+=1
        else:
            c+=1
    result+=raiz
    print(result)
